src/Sulu/Bundle/AdminBundle/Resources/js/components/ColumnList/types.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+export const toolbarItemTypes = {
+    Simple: 'simple',
+    Dropdown: 'dropdown',
+};
+
+export type ButtonConfig = {
+    icon: string,
+    onClick: (string | number) => void,
+};
+
+export type DropdownOptionConfig = {
+    label: string,
+    onClick: (index: string | number) => void,
+    disabled?: boolean,
+};
+
+type ToolbarItemBase = {
+    index: number,
+    icon: string,
+    disabled?: boolean,
+    skin?: 'blue',
+};
+
+export type DropdownProps = ToolbarItemBase & {
+    options: Array<DropdownOptionConfig>,
+};
+
+export type SimpleProps = ToolbarItemBase & {
+    onClick: (index: string | number) => void,
+};
+
+export type DropdownConfig = DropdownProps & { type: typeof toolbarItemTypes.Dropdown };
```

Comment:
I think that check is not working properly, because `typeof` will only check if type is a string instead of the concrete value